modules/gmc-agent/src/main/java/org/gridgain/agent/Agent.java

```diff
@@ -268,7 +270,7 @@ private void connect() {
         nodeConfigurationSrvc = new NodeConfigurationService(ctx, mgr);
         actionSrvc = new ActionService(ctx, mgr);
 
-        execSrvc = Executors.newSingleThreadExecutor();
+        execSrvc = new ThreadPoolExecutor(1, 1, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>());
```

Comment:
Why we can't use one of may `Executors.xxx` methods?

Reply:
Because Executors return FinalizableDelegatedExecutorService and we can't get ThreadPoolExecutor.